engine/sound/src/sound.cpp

```diff
@@ -1555,11 +1570,11 @@ namespace dmSound
         while (free_slots > 0) {
 
             // Get the number of frames available
-            sound->m_FrameCount = sound->m_DeviceFrameCount;
-            uint32_t frame_count = sound->m_FrameCount;
+            uint32_t frame_count = sound->m_DeviceFrameCount;
             if (sound->m_DeviceType->m_GetAvailableFrames)
                 frame_count = sound->m_DeviceType->m_GetAvailableFrames(sound->m_Device);
 
+            sound->m_FrameCount = frame_count;
```

Comment:
Keep the number of mixes frames, so that the Rms/Peak calculations don't go out-of-bounds.